src/main/java/backend/resource/TurboMilestoneDueDateComparator.java

```diff
@@ -0,0 +1,27 @@
+package backend.resource;
+
+
+import java.io.Serializable;
+import java.time.LocalDate;
+import java.util.Comparator;
+
+public class TurboMilestoneDueDateComparator implements Comparator<TurboMilestone>,
+        Serializable {
+
+    @Override
+    public int compare(TurboMilestone m1, TurboMilestone m2) {
+        if (!m1.getDueDate().isPresent() && !m2.getDueDate().isPresent()) {
+            return 0;
+        } else if (!m1.getDueDate().isPresent()) {
+            return 1;
+        } else if (!m2.getDueDate().isPresent()) {
+            return -1;
+        } else {
+            LocalDate m1DueDate = m1.getDueDate().get();
+            LocalDate m2DueDate = m2.getDueDate().get();
+
+            return (m1DueDate.isBefore(m2DueDate)) ? -1 :
+                    (m1DueDate.isEqual(m2DueDate)) ? 0 : 1;
+        }
+    }
```

Comment:
Sorry didn't see this earlier. If yours is merged first then I will replace the similar part with the method. Otherwise, you can help me replace the part.